### ISSUE TITLE
tiny-slider complains about 'clientwidth' upon re-navigating to component in Angular (5.2.9)

### ISSUE THREAD

**Issue by lillacy** [ganlanyuan/tiny-slider#191]
__Issue description__: I'm using tiny-slider with an Angular project at the moment for a horizontal sliding navigation. 

Upon first entering the component where I am importing and using tiny-slider.module, I have no issues. When I navigate to another component and then return, upon resizing I get an error "Cannot read property 'clientWidth' of null". Any idea what is causing this issue?

It doesn't appear to be causing the slider to break, but I only have one slider at the moment.

![image](https://user-images.githubusercontent.com/4213708/39270197-533c0fcc-48a3-11e8-894e-909a4b07aa5e.png)

This is my slider code that I'm using in ngOnInit():
`this.slider = tns({
  container: '.tns-slider',
  controlsContainer: 'tns-slider-controls',
  nav: false,
  controls: false,
  touch: true,
  swipeAngle: false,
  items: 3,
  gutter: 5,
  slideBy: 'page',
  loop: false,
  autoplay: false,
  responsive: {
    992: {
      items: 4
    },
    1200: {
      items: 5
    }
  }
});`

_Tiny-slider version_:   2.6.0
_Browser name && version_:  Chrome 66.0.3359.117
_OS name && version_:   Windows 10

**Comment by ganlanyuan** [comment]
Probably `innerWrapper` became `null`.
https://github.com/ganlanyuan/tiny-slider/blob/0be21926b8994f1f88ca958ab9415e67ac1c71f8/src/tiny-slider.js#L255
Could you make a live demo?

**Comment by lillacy** [comment]
I was able to recreate the issue in CodePen without Angular.

https://codepen.io/lillacy/pen/NMRoEa

Basically Angular is removing the elements from the DOM when you navigate away from the component, and then reinserting them when you reenter.

**Comment by lillacy** [comment]
Hi ganlanyuan,
I managed to clear the complaints to the console. I did not remove the window resize event listener I was using before leaving the page, so even when I tried to use the destroy() method, I was still using the getInfo() method in my listener. So, it was an error on my part. I'm now removing my event listener, then using destroy upon navigating away from the component and it is working as expected.
